clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -624,19 +596,160 @@ struct Table {
+  Table T;
+  if (Error E = T.populateSYCLTable(File))
+    return std::move(E);
+
+  std::vector<module_split::SplitModule> Modules;
+  for (auto &Entry : T.Entries) {
+    llvm::util::PropertySetRegistry Properties;
+    std::string Symbols;
+    if (!Entry.PropFile.empty()) {
+      auto MBOrErr = MemoryBuffer::getFile(Entry.PropFile);
+      if (!MBOrErr)
+        return createFileError(Entry.PropFile, MBOrErr.getError());
+
+      auto &MB = **MBOrErr;
+      auto PropSetOrErr = llvm::util::PropertySetRegistry::read(&MB);
+      if (!PropSetOrErr)
+        return PropSetOrErr.takeError();
+
+      Properties = std::move(**PropSetOrErr);
+    }
+
+    if (!Entry.SymFile.empty()) {
+      auto MBOrErr = MemoryBuffer::getFile(Entry.SymFile);
+      if (!MBOrErr)
+        return createFileError(Entry.SymFile, MBOrErr.getError());
+
+      auto &MB = *MBOrErr;
+      Symbols = std::string(MB->getBufferStart(), MB->getBufferEnd());
+    }
+
+    Modules.emplace_back(Entry.IRFile, std::move(Properties),
+                         std::move(Symbols));
+  }
+
+  return Modules;
+}
+
+// Run sycl-post-link tool
+static Expected<std::vector<module_split::SplitModule>>
+runSYCLPostLinkTool(ArrayRef<StringRef> InputFiles, const ArgList &Args) {
+  Expected<std::string> SYCLPostLinkPath =
+      findProgram("sycl-post-link", {getMainExecutable("sycl-post-link")});
+  if (!SYCLPostLinkPath)
+    return SYCLPostLinkPath.takeError();
+
+  // Create a new file to write the output of sycl-post-link to.
+  auto TempFileOrErr =
+      createOutputFile(sys::path::filename(ExecutableName), "table");
+  if (!TempFileOrErr)
+    return TempFileOrErr.takeError();
+
+  StringRef SYCLPostLinkOptions;
+  if (Arg *A = Args.getLastArg(OPT_sycl_post_link_options_EQ))
+    SYCLPostLinkOptions = A->getValue();
+
+  SmallVector<StringRef, 8> CmdArgs;
+  CmdArgs.push_back(*SYCLPostLinkPath);
+  SYCLPostLinkOptions.split(CmdArgs, " ", /* MaxSplit = */ -1,
+                            /* KeepEmpty = */ false);
+  CmdArgs.push_back("-o");
+  CmdArgs.push_back(*TempFileOrErr);
+  for (auto &File : InputFiles)
+    CmdArgs.push_back(File);
+  if (Error Err = executeCommands(*SYCLPostLinkPath, CmdArgs))
+    return std::move(Err);
+
+  return parseSplitModulesFromFile(*TempFileOrErr);
+}
+
+Error printSplitModules(const std::vector<module_split::SplitModule> &Modules) {
+  for (size_t I = 0, E = Modules.size(); I != E; ++I) {
+    SMDiagnostic Err;
+    LLVMContext C;
+    std::unique_ptr<Module> M = parseIRFile(Modules[I].ModuleFilePath, Err, C);
+    if (!M) {
+      std::string Msg;
+      raw_string_ostream S(Msg);
+      Err.print(ExecutableName.begin(), S);
+      return createStringError(inconvertibleErrorCode(), Msg);
+    }
+
+    errs() << formatv("Split Module. Index: {0}\n", I);
+    M->dump();
+  }
+
+  return Error::success();
+}
+
+static Expected<std::vector<module_split::SplitModule>>
+runSYCLSplitLibrary(ArrayRef<StringRef> InputFiles, const ArgList &Args,
```

Comment:
We will want to parse the -sycl-post-link-options here.

Reply:
Can you please share your thoughts on this? Thanks

Reply:
The issue here is that the library is not complete. The plan is to fulfill missing functionality and add the corresponding splitting option parsing.